### PR TITLE
Security RecordTrak ContextActions menu

### PR DESCRIPTION
**Pull request by CheffieGithub**
[QOL]
## About the PR 
Changes the security RecordTrak to use a ContextActions menu which doesn't close when moving.

image
## Why's this needed? 
Ease of use and a slight bit faster to change modes on average.

## Changelog 

```changelog
(u)Cheffie
(*)The security RecordTrak now uses a context actions menu.
```

### CODE REVIEW

File: code/obj/item/device/scanners.dm [hunk 1]
```diff
@@ -700,22 +723,30 @@ that cannot be itched
...
+				boutput(user, "<span class='notice'>you switch the record mode to None</span>")
+
+			if(PRISONER_MODE_PAROLED)
+				boutput(user, "<span class='notice'>you switch the record mode to Paroled</span>")
+
+			if(PRISONER_MODE_RELEASED)
+				boutput(user, "<span class='notice'>you switch the record mode to Released</span>")
+
+			if(PRISONER_MODE_INCARCERATED)
+				boutput(user, "<span class='notice'>you switch the record mode to Incarcerated</span>")
```
Comment (frawhst) [inline code/obj/item/device/scanners.dm:743]:
```suggestion
			if(PRISONER_MODE_NONE)
				boutput(user, "<span class='notice'>you switch the record mode to None.</span>")

			if(PRISONER_MODE_PAROLED)
				boutput(user, "<span class='notice'>you switch the record mode to Paroled.</span>")

			if(PRISONER_MODE_RELEASED)
				boutput(user, "<span class='notice'>you switch the record mode to Released.</span>")

			if(PRISONER_MODE_INCARCERATED)
				boutput(user, "<span class='notice'>you switch the record mode to Incarcerated.</span>")
```
These were wrong before but might as well

File: code/obj/item/device/scanners.dm [hunk 2]
```diff
@@ -700,22 +723,30 @@ that cannot be itched
...
 
 		add_fingerprint(user)
 		return
+	dropped(var/mob/user)
+		. = ..()
+		user.closeContextActions()
```
Comment (frawhst) [inline code/obj/item/device/scanners.dm:749]:
```suggestion
		return
		
	dropped(var/mob/user)
		. = ..()
		user.closeContextActions()
```

File: code/obj/item/device/scanners.dm [hunk 3]
```diff
@@ -612,13 +612,27 @@ that cannot be itched
...
 	w_class = W_CLASS_NORMAL
 	item_state = "recordtrak"
 	flags = FPRINT | TABLEPASS | ONBELT | CONDUCT | EXTRADELAY
 	mats = 3
+	contextLayout = new /datum/contextLayout/experimentalcircle
+
+	///List of record settings
+	var/list/modes = list(PRISONER_MODE_NONE, PRISONER_MODE_PAROLED, PRISONER_MODE_INCARCERATED, PRISONER_MODE_RELEASED)
+	///The current setting
+	var/mode = 1
```
Comment (frawhst) [inline code/obj/item/device/scanners.dm:626]:
```suggestion
	var/mode = PRISONER_MODE_NONE
```

File: code/obj/item/device/scanners.dm [hunk 4]
```diff
@@ -612,13 +612,27 @@ that cannot be itched
...
 	desc = "A device used to scan in prisoners and update their security records."
 	icon_state = "recordtrak"
-	var/mode = 1
 	var/datum/db_record/active1 = null
 	var/datum/db_record/active2 = null
 	w_class = W_CLASS_NORMAL
 	item_state = "recordtrak"
 	flags = FPRINT | TABLEPASS | ONBELT | CONDUCT | EXTRADELAY
 	mats = 3
+	contextLayout = new /datum/contextLayout/experimentalcircle
```
Comment (frawhst) [inline code/obj/item/device/scanners.dm:621]:
```suggestion
```
I don't know why this compiles but it leads to weird behavior, so just instantiate the thing in `New()`

**Review by frawhst — COMMENTED** [review]
good stuff

File: code/obj/item/device/scanners.dm (hunk 4 above)
Comment (CheffieGithub) [reply code/obj/item/device/scanners.dm]: the context layout is made the same way as #9199 so if it causes strange behaviour that should be changed also

Comment (frawhst) [reply code/obj/item/device/scanners.dm]: Oh I didn't notice that. and I merged that!!! Probably fine then

**Review by frawhst — APPROVED** [review]

**Review by pali6 — APPROVED** [review]

**Pull request merged**